node/stateSync.go

```diff
@@ -171,10 +172,14 @@ func (s *stateSync) run() error {
 
 func (s *stateSync) startWorkerThread() {
 	var workerId uint32
-	peersNum := len(s.peers)
-	for workerId = 0; workerId < uint32(peersNum*concurrentSyncRequestPerNeighbor); workerId++ {
+	peersNum := uint32(len(s.peers))
+	workerNum := config.Parameters.SyncStateMaxPeer
+	if workerNum > peersNum {
+		workerNum = peersNum
+	}
+	for workerId = 0; workerId < workerNum; workerId++ {
 		go func(workerId uint32) {
-			s.startWorker(s.peers[workerId%uint32(peersNum)])
+			s.startWorker(s.peers[workerId%workerNum])
```

Comment:
Should be `peersNum` just in case `workerNum > peersNum` in the future

Reply:
Fixed